src/evalfunc.c

```diff
@@ -768,7 +768,7 @@ static funcentry_T global_functions[] =
     {"range",		1, 3, FEARG_1,	  ret_list_number, f_range},
     {"readdir",		1, 2, FEARG_1,	  ret_list_string, f_readdir},
     {"readdirex",	1, 2, FEARG_1,	  ret_list_dict_any, f_readdirex},
-    {"readfile",	1, 3, FEARG_1,	  ret_any,	f_readfile},
+    {"readfile",	1, 3, FEARG_1,	  ret_list_string,	f_readfile},
```

Comment:
Quoting `:help readfile()`:
```
readfile({fname} [, {type} [, {max}]])
…
When {type} contains "B" a |Blob| is returned with the binary
data of the file unmodified.
```
So if it can return a Blob, I suppose that putting `ret_list_string` is wrong.
But I don't know the consequences of putting `ret_any` or `ret_list_string`.

Reply:
Need to create a function like `ret_f_function()` or `ret_f_getline()`.